### PR TITLE
include docker 18.06.1 missed dependency

### LINKED ISSUE #6225: Docker 17.06+ won't install on CentOS / RHEL / Amazon Linux due to unavailable dependency
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Kops head shortly after the 1.11.beta release ( Version 1.11.0-alpha.1 (git-4806708) ), which included my PR https://github.com/kubernetes/kops/commit/4806708c3bd6e8876cfe466de51d2092ad59e71b
**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
1.11.5
**3. What cloud provider are you using?**
AWS
**4. What commands did you run?  What is the simplest way to reproduce this issue?**
Create a cluster, set docker version in cluster spec to anything higher than 17.06 (I tried 18.06 and 17.09 and use a RHEL / CentOS / Amazon Linux 2 AMI. (I personally use Amazon Linux 2)
As of 17.06 docker decided to use `container-selinux` as dependency instead of packaging in their own, so any version higher than that will show this issue.
If you don't even want to bother with kops, launching one of those AMI's and manually installing docker from an RPM also has this issue.
**5. What happened after the commands executed?**
Nodeup tries to install the higher docker version, but the instal fails with the following message:
```
Error: Package: docker-ce-18.06.1.ce-3.el7.x86_64 (/docker-ce-18.06.1.ce-3.el7.x86_64)
           Requires: container-selinux >= 2.9
```
Which is not present by default in these distros. Centos and RHEL offer it in their extras I believe (that you might need to enable), Amazon linux does not have it available. You can install it manually from http://mirror.centos.org/centos/7/extras/x86_64/Packages/ though. Installing docker through the extras channels in these distros does not require this step. 

Since Docker can't be installed, the cluster obviously does not come up.
**6. What did you expect to happen?**
Cluster comes up with the requested version of docker

**9. Anything else do we need to know?**
I did my PR mirroring what was already available for 17.09, not expecting any issues but didn't verify that 17.09 itself would work. I am not really sure how to handle this case since the dependency is not readily availably from yum and fixing it seems to depend on which distro you are using. 
Should I add a direct URL to the dependency (fixing the version, which is not ideal), bake this into the AMI (not very future proof either) or should this be solved somewhere else in nodeup? 

The install issue itself is a known issue outside of kops fwiw: https://stackoverflow.com/questions/45272827/docker-ce-on-rhel-requires-container-selinux-2-9

### PR DESCRIPTION
**Pull request by nareshku**
Fixes #6225 

Thanks to @[xrl](https://github.com/xrl) for doing the actual work.

**Testing Completed with docker v17.09.0**
Created a cluster on Rhel7 with Calico and deleted

**Testing Completed with docker v18.06.1-ce**
Created a cluster on Rhel7 with Calico and deleted
Created a cluster on AWS Linux 2 with Amazon VPC CNI and deleted

### CODE REVIEW

File: nodeup/pkg/model/docker.go [hunk 1]
```diff
@@ -540,7 +540,17 @@ var dockerVersions = []dockerVersion{
 		Dependencies:  []string{"bridge-utils", "libapparmor1", "libltdl7", "perl"},
 	},
 
-	// 18.06.1 - CentOS / Rhel7
+	// 18.06.1 - CentOS / Rhel7 (two packages)
+	{
+		DockerVersion: "18.06.1",
+		Name:          "container-selinux-2",
+		Distros:       []distros.Distribution{distros.DistributionRhel7, distros.DistributionCentos7},
```
Comment (xrl) [inline nodeup/pkg/model/docker.go]: Hmm, I think I messed up the formatting here, there should be a newline between `DistributionCentos7` and the `}`.

**Comment by bcorijn** [comment]
@nareshku I would add this for Docker 17.09 as well, since this fails with the same issue. 
Only downside I see to this approach is that it fixes the version number.

**Comment by nareshku** [comment]
@bcorijn I agree. I have included the dependency for docker v17.09 as well.

**Comment by nareshku** [comment]
@justinsb @mikesplain We have been waiting for docker v18.06 support for sometime, it would be great if you can help us with a release v1.11.1 including this change.

**Comment by justinsb** [comment]
I think the locking to a particular version is a concern; mitigating that is that we're locking the docker package anyway.  We can at least do something, except that there's no way to express a newer version of container-se because we're currently running off the docker version alone.

I think this gets better with containerd, which isn't too far away now, and will hopefully be packaged more completely.

/approve
/lgtm

**Comment by justinsb** [comment]
Candidate for 1.11.1 I think

**Comment by bcorijn** [comment]
@justinsb Can the PR that adds 18.06 be cherry-picked as well in that case https://github.com/kubernetes/kops/pull/6202 ? I believe that didn't make it into 1.11 either

**Comment by justinsb** [comment]
Good point @bcorijn  - tagging it also :-)

**Pull request merged**